src/components/legend/attributes.js

```diff
@@ -77,6 +77,16 @@ module.exports = {
             'or remain *constant* independent of the symbol size on the graph.'
         ].join(' ')
     },
+    itemsymbol: {
+        valType: 'enumerated',
+        values: ['trace', 'constant'],
```

Comment:
I can imagine people may want to have `square` or other symbols in future.
So it might be better to use `circle` instead of `constant` here.

Reply:
Sure. I just figured I would err on the side of making it work similar to how `itemsizing` works. But obviously switching the option to "circle" would be simple.

Reply:
Also maybe worth noting that if #3735 is ever implemented to expose basically all marker options for the legend, I imagine folks would use that more extensive customization of the legend symbol.

Reply:
I like `circle` :)

Reply:
I switched atributes.js to list the 'circle' option. But in the style.js file, I believe I have set it up to use any valid symbol the user enters for `itemsymbol`. Meaning the docs could be expanded to list all valid symbol names as possible values. I'm just not sure where such a list would come from. For now, attributes.js just mentions 'circle'.

Reply:
FYI - The list is coming from here: https://github.com/plotly/plotly.js/blob/master/src/components/drawing/symbol_defs.js

Reply:
I updated it to pull in the list of symbols the same way it is done for the scatter trace attributes. I believe that should supply the full list of symbol options.

Reply:
Just thought I'd bump this. I believe I have it set up so it will now accept any symbol name (not just circle) to use as the legend symbol. Anything else left to do before this could be merged in?

Reply:
The tests are failing at the moment.
Could you please sync the master branch of your fork with upstream/master and then merge master into your branch?

Reply:
hmm I pulled from the plotly master to my master, then rebased my feature branch the updated master. Something went wonky though, the PR says there are 462 changes. I'll try to figure out what I did wrong.

Reply:
I think I fixed the wonky merge. Back to just the 3 changed files.